Frameworks/UIKit/UIPasteboard.mm

```diff
@@ -238,21 +344,23 @@ - (id)valueForPasteboardType:(NSString*)pasteboardType {
  @Notes only NSString and UIImage are supported.
 */
 - (void)setData:(NSData*)data forPasteboardType:(NSString*)pasteboardType {
-    if (data == nil || pasteboardType == nil) {
+    if (![_name isEqualToString:UIPasteboardNameGeneral]) {
+        UNIMPLEMENTED_WITH_MSG("Only general pasteboard supported");
         return;
     }
 
-    // Copy and Paste
-    if ([_name isEqualToString:UIPasteboardNameGeneral]) {
-        WADDataPackage* dataPackage = [[WADDataPackage make] autorelease];
-        dataPackage.requestedOperation = WADDataPackageOperationCopy;
+    if (data && pasteboardType) {
+        // only set data when both data and pasteType are avaiable
+        StrongId<WADDataPackage> dataPackage = [WADDataPackage make];
```

Comment:
you could also ping Dustin to confirm

---
In reply to: [87678935](https://github.com/Microsoft/WinObjC/pull/1358#discussion_r87678935) [](ancestors = 87678935)